### ISSUE TITLE
Make input remappable.

### ISSUE THREAD

**Issue by NogginBops** [TeamPorcupine/ProjectPorcupine#13]
This will allow for inputs to be dynamically changed and remapped.

**Comment by svmnotn** [comment]
From what I remember of the Input Manager it is difficult to use properly at runtime (As if you are the user then the only way to change things is at the screen that launches the game); thus unless I'm misremembering or they have updated it, we might want to look at making something that just relies on `Input.GetKey`.

**Comment by NogginBops** [comment]
That is true. There might be some other libraries that could be used or a totally custom solution could be built.

**Comment by quill18** [comment]
Step one is probably to have things that have hotkeys.  Right now there's just left/right click and escape -- not the sort of thing that needs to be remapped since it's pretty darned straightforward and obvious.

So we should figure out (as a separate issue maybe?) what kind of things might need hotkeys, how we want to use them, and how we want to present them.  Maybe toggling modes on characters/objects (turn things on & off, marking things for deconstructing, etc...).

Step two will be to determine if there's actual value in changing the keys.  Are there other base builders  that allow key remapping? What's their interface like? Does anyone use the remapping function?

**Comment by AaronBuckles** [comment]
We definitely need to be able to delete an object with a hotkey.  This should be customizable because some people may want only one hand on the keyboard, while others may want it to be something like 'B' so they don't accidentally press it.

**Comment by emilkris33** [comment]
I think all this is probably a bit to early. I agree that we need rebindable hotkeys, but before we can do that meaningfully we need to have a better structure to the building UI.

**Comment by NogginBops** [comment]
I think I will close this issue for now. We could open it when this becomes relevant in the future.
